migrations/src/postDeploymentSetup.js

```diff
@@ -127,12 +131,38 @@ module.exports = async () => {
   pendingTransactions.push(await assetManager.upgradeDelegator(delegatorAddress));
   pendingTransactions.push(await governance.grantRole(GOVERNER_ROLE, signers[0].address));
 
-  // eslint-disable-next-line no-console
   console.log('Waiting for post-deployment setup transactions to get confirmed');
   for (let i = 0; i < pendingTransactions.length; i++) {
     pendingTransactions[i].wait();
   }
 
-  // eslint-disable-next-line no-console
+  const jobs = await getJobs();
+  const collections = await getCollections();
+  console.log('Creating Jobs');
+
+  for (let i = 0; i < jobs.length; i++) {
+    const job = jobs[i];
+    await assetManager.createJob(job.weight, job.power, job.selectorType, job.name, job.selector, job.url);
+    console.log(`Job Created :  ${job.name}`);
+  }
+
+  console.log('Creating Collections');
+  console.log('Waiting for Confirm state : 4.......');
+  let state = 0;
+  const numStates = await stakeManager.NUM_STATES();
+  const stateLength = (BigNumber.from(await stakeManager.epochLength())).div(numStates);
+
+  while (state !== 4) {
+    const blockNumber = await ethers.provider.getBlockNumber();
+    state = Number(((BigNumber.from(blockNumber)).div(stateLength)).mod(numStates));
+    console.log('Current state', state);
+    await sleep(10000);
+  }
+
+  for (let i = 0; i < collections.length; i++) {
+    const collection = collections[i];
+    await assetManager.createCollection(collection.jobIDs, collection.aggregationMethod, collection.power, collection.name);
```

Comment:
Should check for confirm state before creating each collection, and wait for confirm state if not in confirm state.

Reply:
resolved https://github.com/razor-network/contracts/pull/579/commits/57925d7b53c6d85e2116b416953ea4d30159789e